src/CloneExtensions/Deipax/FieldInfoExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Reflection;
+
+namespace Deipax.Core.Extensions
+{
+    public static class FieldInfoExtensions
+    {
+        public static bool IsBackingField(this FieldInfo source, bool defaultValue = false)
+        {
+            if (source != null)
+            {
+                return source.Name.IndexOf(">k__BackingField", 0) >= 0;
```

Comment:
Use `string.Contains` instead.

Reply:
"Can we achieve the same performance increase without that extra layer?" - Marcin 

Likely we can, yes but it would require new, untested, code.  While this code is new to you, I have been using it for quite awhile in my own work.  The reason I chose a class to collect the information and cache it in one place is because there are non-trivial issues when dealing with private properties/fields and I want access to this information in more than one place.

The reason I left this speed tweak for last was because it was a much larger change, it is not necessary.  If you want to leave it out because of possible instability, that is understandable. Perhaps if you created a new branch where I could add this work along with other more unstable changes ... such as support for polymorphism, support for attribute driven directives for cloning or a method to control cloning for library classes (classes authored by another), a testing framework for performance comparison just to name a few.

Let me know what you think.  If you want to pull a second branch, I can do the work there.  If you decide you want the code in this branch, then I will make the changes you have requested.

"Use string.Contains instead." - Marcin

I can do it, no problem, if you decide to incorporate the change into this branch ... but I am curious as to why.

Reply:
It not that I'm afraid of instability, it's just that I don't see how caching type system information in `ConcurrentDictionary` helps if you should never ask for data on the same type more than once (because for each cloned type the clone delegate will be generated just once). Let me see if I can get the same end-result without that extra layer, as I love the perf wins here. Just not a huge fun of all that complexity which even though might be useful in other application does not seem to add value in this particular case.

About `Contains`, it's just more readable.